library/source/OpenCDMSystemPrivate.cpp

```diff
@@ -124,3 +124,29 @@ bool OpenCDMSystemPrivate::deleteDrmStore() const
     }
     return m_cdmBackend->deleteDrmStore();
 }
+
+bool OpenCDMSystemPrivate::getMetricSystemData(uint32_t *bufferLength, uint8_t *buffer) const
+{
+    if (!m_cdmBackend || !bufferLength)
+    {
+        return false;
+    }
+
+    std::vector<uint8_t> tempBuffer;
+
+    if (!m_cdmBackend->getMetricSystemData(tempBuffer))
+    {
+        return false;
+    }
+
+    if (*bufferLength < tempBuffer.size())
```

Comment:
this check not needed, if you use vector. This function should look like this:

if (!m_cdmBackend)
{
    return false;
}
return m_cdmBackend->getMetricSystemData(bufferFromFunctionArg);